types/carbon__icons/index.d.ts

```diff
@@ -0,0 +1,57 @@
+// Type definitions for @carbon/icons 10.29
+// Project: https://github.com/carbon-design-system/carbon/tree/master/packages/icons
+// Definitions by: Eric Liu <https://github.com/metonym>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.0
+
+declare module "@carbon/icons" {
+    type CarbonIconSize = 16 | 20 | 24 | 32;
```

Comment:
I believe you can unwrap these types and just export them. Does that not work?

Reply:
I tried that initially but received an error when using it in the `declare module "@carbon/icons/metadata.json"` ambient module.

Reply:
You don’t need to import them at that point because they’ll be in scope.

Reply:
Actually, I believe you’ll have to move the declaration for metadata.json to its own file. Then they should play nicely together. (Don’t forget to add `metadata.d.ts` to  `files` in tsconfig.json.)

Reply:
@andrewbranch thank you for the suggestion – moving the declaration to a separate file works. The lint/test scripts passed locally but I'm seeing this [error in the CI](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52154/checks?check_run_id=2256184704). Any suggestions?

```
Error: carbon__icons: Only index.d.ts may be listed explicitly in tsconfig's "files" entry.
Other d.ts files must either be referenced through index.d.ts, tests, or added to OTHER_FILES.txt.
```

Reply:
Applied the same fix to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52155